Add support for Zigbee fota with XiP & new Zigbee weather station + SMP server application

sebastiandraus opened the pull request.

sebastiandraus committed:
Allow enabling Zigbee FOTA when XiP code relocation is used

sebastiandraus committed:
Enable Zigbee FOTA in Zigbee Weather station application

sebastiandraus committed:
Add Zigbee weather station application merged with SMP srv app

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/NordicPlayground/ncs-nrf5340-xip-app?pullRequest=10) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/NordicPlayground/ncs-nrf5340-xip-app?pullRequest=10) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/NordicPlayground/ncs-nrf5340-xip-app?pullRequest=10) it.</sub>

nordicjm commented on modules/mcuboot/CMakeLists.txt:
```diff
@@ -790,21 +790,39 @@ if (CONFIG_DFU_MULTI_IMAGE_PACKAGE_BUILD)
+      list(APPEND dfu_multi_image_ids 0)
+      list(APPEND dfu_multi_image_paths "${PROJECT_BINARY_DIR}/internal_flash_update.bin")
+      list(APPEND dfu_multi_image_targets mcuboot_sign_target)
+    endif()
```
QSPI part should be moved inside this if()

nordicjm commented on zigbee_weather_station/configuration/thingy53_nrf5340_cpuapp/prj.conf:
```diff
@@ -51,3 +51,45 @@ CONFIG_BUILD_NO_GAP_FILL=y
+CONFIG_ZIGBEE_FOTA_HW_VERSION=52
+CONFIG_ZIGBEE_FOTA_ENDPOINT=5
+CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION="1.2.3"
+# CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION="1.2.4"
```
Duplicate line can go

nordicjm commented on zigbee_weather_station/configuration/thingy53_nrf5340_cpuapp/prj.conf:
```diff
@@ -51,3 +51,45 @@ CONFIG_BUILD_NO_GAP_FILL=y
+
+#######################
+# Zigbee FOTA board overlay #
+#######################
```
Header size not aligned

nordicjm commented on zigbee_weather_station/configuration/thingy53_nrf5340_cpuapp/prj_release.conf:
```diff
@@ -47,3 +47,45 @@ CONFIG_BUILD_NO_GAP_FILL=y
+CONFIG_ZIGBEE_FOTA_HW_VERSION=52
+CONFIG_ZIGBEE_FOTA_ENDPOINT=5
+CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION="1.2.3"
+# CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION="1.2.4"
```
As above

nordicjm commented on zigbee_weather_station/configuration/thingy53_nrf5340_cpuapp/prj_release.conf:
```diff
@@ -47,3 +47,45 @@ CONFIG_BUILD_NO_GAP_FILL=y
+CONFIG_NRF_802154_RX_BUFFERS=32
+
+
+#######################
+# Zigbee FOTA board overlay #
+#######################
```
As above

nordicjm commented on zigbee_weather_station_mcumgr/CMakeLists.txt:
```diff
@@ -0,0 +1,104 @@
+zephyr_code_relocate(LIBRARY drivers__sensor__bme680 LOCATION EXTFLASH_TEXT NOCOPY)
+zephyr_code_relocate(LIBRARY drivers__sensor__bmm150 LOCATION EXTFLASH_TEXT NOCOPY)
+zephyr_code_relocate(LIBRARY ..__nrf__lib__dk_buttons_and_leds LOCATION EXTFLASH_TEXT NOCOPY)
+# zephyr_code_relocate(LIBRARY subsys__net LOCATION EXTFLASH_TEXT NOCOPY)
```
Commented line can go

nordicjm commented on zigbee_weather_station_mcumgr/README.rst:
```diff
@@ -0,0 +1,541 @@
+Building and running
+********************
+
+.. |application path| replace:: :file:`applications/zigbee_weather_station`
```
Paths should probably be updated

nordicjm commented on zigbee_weather_station_mcumgr/pm_static.yml:
```diff
@@ -0,0 +1,127 @@
+  device: MX25R64
+  end_address: 0x160000
+  region: external_flash
+  size: 0x40000
```
Just noticed that this is not in sync with the linker file here on in the one it was copied from

sebastiandraus committed:
zigbee: zigbee_weather_station_mcumgr: Align configuration

This commit aligns configuration of the sample with changes introduced
with upcoming NCS release v.2.5

Signed-off-by: Draus, Sebastian <sebastian.draus@nordicsemi.no>

sebastiandraus committed:
cmakelists: Generate Image ID 2 (split APP core img) when XiP is enabled

sebastiandraus committed:
zigbee_weather_station: Cleanup project configuration files

sebastiandraus committed:
zigbee_weather_station_mcumgr: Cleanup CMakeLists.txt file

sebastiandraus committed:
zigbee_weather_station_mcumgr: Align address in linker file

This commit changes the length of the external flash memory
in the linker file as suggested by Jamie.

sebastiandraus committed:
zigbee_weather_station_mcumgr: Align path to app in README.rst

nordicjm commented on modules/mcuboot/CMakeLists.txt:
```diff
@@ -790,21 +790,39 @@ if (CONFIG_DFU_MULTI_IMAGE_PACKAGE_BUILD)
+    list(APPEND dfu_multi_image_ids 2)
+    list(APPEND dfu_multi_image_paths "${PROJECT_BINARY_DIR}/qspi_flash_update.bin")
+
+
```
Excess whitespace can go

nordicjm reviewed (commented):
Comments on this PR are still outstanding

sebastiandraus committed:
modules: mcuboot: Cleanup CMakeLists whitespace

sebastiandraus committed:
zigbee_weather_station_mcumgr: Fix pm_static file

This commit changes the opm_static file to avoid overlapping
partitions in FLASH.

sebastiandraus commented:
Rebased at newest main branch in ncs-playground and removed pointing to PR in nrf repo so this ISN'T WORKING right now.
Aligning nrf repo revision in progress
